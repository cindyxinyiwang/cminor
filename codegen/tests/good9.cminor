/*
if statements
*/

main: function integer ( argc: integer, argv: array [] string ) =
{
	i: boolean = true;
	j: boolean = false;
	k: boolean;
	
	if (j)
		print "wrong if\n";
	else
		if (j)
			print "wrong if\n";
		else
			if (i)
				print "correct if\n";
			else
				print "wrong if\n";
	my_int: integer = 0;
	if (45 > 20) {
		my_int++;
	} else {
		my_int--;
	}
	if (my_int == 1)
		print "correct if\n";
	return 0;
}
