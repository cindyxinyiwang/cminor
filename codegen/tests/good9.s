.text
.globl main
main:
############# preamble before function
pushq %rbp
movq %rsp, %rbp
subq $0, %rsp	#allocate local variables
pushq %rbx
pushq %r12
pushq %r13
pushq %r14
pushq %r15
####### body of function
MOV $.LS0, %r8
MOV %r8, %rdi	 # first argument
MOVQ $0, %rax	 #there are zero floating point args
PUSHQ %r10
PUSHQ %r11
CALL print_string
POPQ %r10
POPQ %r11
MOV $.LS1, %r8
MOV %r8, %rdi	 # first argument
MOVQ $0, %rax	 #there are zero floating point args
PUSHQ %r10
PUSHQ %r11
CALL print_string
POPQ %r10
POPQ %r11
############# postamble after function
popq %r15
popq %r14
popq %r13
popq %r12
popq %rbx
movq %rbp, %rsp
popq %rbp
ret
.data
.LS0:
.string "hello"
.LS1:
.string "world\n"
