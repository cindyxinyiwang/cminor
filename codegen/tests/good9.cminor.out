size resovles to global size
size resovles to global size
size resovles to global size
size resovles to global size
size resovles to global size
a resovles to param 0
a resovles to param 0
a resovles to param 0
arr2 resovles to global arr2
func resovles to global func
